src/clr-angular/utils/drag-and-drop/if-dragged.spec.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component, ElementRef, ViewChild} from "@angular/core";
+import {TestBed} from "@angular/core/testing";
+
+import {ClrDraggable} from "./draggable";
+import {emulateDragEvent} from "./helpers.spec";
+import {ClrIfDragged} from "./if-dragged";
+import {ClrDragAndDropEventBus} from "./providers/drag-and-drop-event-bus";
+
+export default function(): void {
+    describe("ClrIfDragged", function() {
+        beforeEach(function() {
+            TestBed.configureTestingModule(
+                {declarations: [IfDraggedTest, ClrIfDragged, ClrDraggable], providers: [ClrDragAndDropEventBus]});
+
+            this.fixture = TestBed.createComponent(IfDraggedTest);
+            this.fixture.detectChanges();
+            this.testComponent = this.fixture.componentInstance;
+            this.testElement = this.fixture.nativeElement;
+            this.draggableEl = this.fixture.componentInstance.draggableEl.nativeElement;
+        });
+
+        afterEach(function() {
+            this.fixture.destroy();
+        });
+
+        it("should not display anything on normal state", function() {
+            expect(this.draggableEl.textContent.trim()).toBe("");
+        });
+
+        it("should not display anything on mousemove or touchstart events", function() {
+            emulateDragEvent("mousedown", 0, 0, this.draggableEl);
```

Comment:
You're actually testing `clrDraggable` all over again here, which is problematic. You shouldn't have a  `clrDraggable` element at all in this test, just the `*clrIfDragged` directive, and you should simply manipulate the service directly. You can declare the service as a root provider in the testing module, it works absolutely fine for this spec.

Reply:
But if the requirement is I shouldn't have a `clrDraggable` element at all in the test, I can manage it :P